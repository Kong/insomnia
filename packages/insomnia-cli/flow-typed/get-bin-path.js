// @flow

declare module 'get-bin-path' {
  declare module.exports: *;
}
