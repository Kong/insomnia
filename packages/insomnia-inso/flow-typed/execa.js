// @flow

declare module 'execa' {
  declare module.exports: *;
}
