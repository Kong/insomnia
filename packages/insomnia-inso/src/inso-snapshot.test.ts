import execa from 'execa';
import { getBinPathSync } from 'get-bin-path';

// MAKE SURE YOU BUILD THE PROJECT BEFORE RUNNING THESE TESTS.
// These tests use the executable /bin/inso, which relies on /dist.
describe('Snapshot for', () => {
  it.each([
    '-h',
    '--help',
    'help',
    'generate -h',
    'generate config -h',
    'run -h',
    'run test -h',
    'lint -h',
    'lint spec -h',
    'export -h',
    'export spec -h',
  ])(
    '"inso %s"',
    async (args) => {
      const binPath = getBinPathSync();
      if (binPath === undefined) {
        throw new Error('unable to get binary path');
      }

      const { stdout } = await execa(binPath, args.split(' '));
      expect(stdout).toMatchSnapshot();
    },
    30000,
  );
});
