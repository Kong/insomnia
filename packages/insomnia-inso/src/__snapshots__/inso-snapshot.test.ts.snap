// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for "inso --help" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version           output the version number
  -w, --workingDir <dir>  set working directory
  -a, --appDataDir <dir>  set the app data directory
  --config <path>         path to configuration file
  --verbose               show additional logs while running the command
  --printOptions          print the loaded options
  --ci                    run in CI, disables all prompts
  -h, --help              display help for command

Commands:
  generate                Code generation utilities
  run                     Execution utilities
  lint                    Linting utilities
  export                  Export data from insomnia models
  script <name>           Run scripts defined in .insorc
  help [command]          display help for command"
`;

exports[`Snapshot for "inso -h" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version           output the version number
  -w, --workingDir <dir>  set working directory
  -a, --appDataDir <dir>  set the app data directory
  --config <path>         path to configuration file
  --verbose               show additional logs while running the command
  --printOptions          print the loaded options
  --ci                    run in CI, disables all prompts
  -h, --help              display help for command

Commands:
  generate                Code generation utilities
  run                     Execution utilities
  lint                    Linting utilities
  export                  Export data from insomnia models
  script <name>           Run scripts defined in .insorc
  help [command]          display help for command"
`;

exports[`Snapshot for "inso export -h" 1`] = `
"Usage: inso export [options] [command]

Export data from insomnia models

Options:
  -h, --help                   display help for command

Commands:
  spec [options] [identifier]  Export an API Specification to a file
  help [command]               display help for command"
`;

exports[`Snapshot for "inso export spec -h" 1`] = `
"Usage: inso export spec [options] [identifier]

Export an API Specification to a file

Options:
  -o, --output <path>  save the generated config to a file
  -h, --help           display help for command"
`;

exports[`Snapshot for "inso generate -h" 1`] = `
"Usage: inso generate [options] [command]

Code generation utilities

Options:
  -h, --help                     display help for command

Commands:
  config [options] [identifier]  Generate configuration from an api spec.
  help [command]                 display help for command"
`;

exports[`Snapshot for "inso generate config -h" 1`] = `
"Usage: inso generate config [options] [identifier]

Generate configuration from an api spec.

Options:
  -t, --type <value>   type of configuration to generate, options are
                       [declarative, kubernetes] (default: declarative)
  --tags <tags>        comma separated list of tags to apply to each entity
  -o, --output <path>  save the generated config to a file
  -h, --help           display help for command"
`;

exports[`Snapshot for "inso help" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version           output the version number
  -w, --workingDir <dir>  set working directory
  -a, --appDataDir <dir>  set the app data directory
  --config <path>         path to configuration file
  --verbose               show additional logs while running the command
  --printOptions          print the loaded options
  --ci                    run in CI, disables all prompts
  -h, --help              display help for command

Commands:
  generate                Code generation utilities
  run                     Execution utilities
  lint                    Linting utilities
  export                  Export data from insomnia models
  script <name>           Run scripts defined in .insorc
  help [command]          display help for command"
`;

exports[`Snapshot for "inso lint -h" 1`] = `
"Usage: inso lint [options] [command]

Linting utilities

Options:
  -h, --help         display help for command

Commands:
  spec [identifier]  Lint an API Specification
  help [command]     display help for command"
`;

exports[`Snapshot for "inso lint spec -h" 1`] = `
"Usage: inso lint spec [options] [identifier]

Lint an API Specification

Options:
  -h, --help  display help for command"
`;

exports[`Snapshot for "inso run -h" 1`] = `
"Usage: inso run [options] [command]

Execution utilities

Options:
  -h, --help                   display help for command

Commands:
  test [options] [identifier]  Run Insomnia unit test suites
  help [command]               display help for command"
`;

exports[`Snapshot for "inso run test -h" 1`] = `
"Usage: inso run test [options] [identifier]

Run Insomnia unit test suites

Options:
  -e, --env <identifier>         environment to use
  -t, --testNamePattern <regex>  run tests that match the regex
  -r, --reporter <reporter>      reporter to use, options are [dot, list, min,
                                 progress, spec] (default: spec)
  -b, --bail                     abort (\\"bail\\") after first test failure
  --keepFile                     do not delete the generated test file
  --disableSSL                   disable SSL validation
  -h, --help                     display help for command"
`;
